packages/ui-kit/src/helpers/themeUtils.ts

```diff
@@ -0,0 +1,61 @@
+import type { ThemeTokenProps } from '../themes/theme.types'
+import { themeTokens } from '../themes/themeTokens'
+import { camelCaseToKebabCase } from './camelCaseToKebabCase'
+
+const themeDict = themeTokens.map((themeToken) => ({
+  name: themeToken,
+  cssVarName: `--propel-${camelCaseToKebabCase(themeToken)}`
+}))
+
+/**
+ * Parses the computed style of a given HTML element and extracts theme-related properties.
+ * It maps CSS variable names to their corresponding theme token names and retrieves their values.
+ * This is useful for dynamically reading the current theme values from an element.
+ *
+ * @param {HTMLElement} themeContainer - The HTML element from which to parse the theme.
+ * @returns {Partial<ThemeTokenProps>} An object containing the theme properties and their values.
+ */
+export const parseComputedStyle = (themeContainer: HTMLElement) => {
+  const computedStyle = getComputedStyle(themeContainer)
+  const theme: Partial<ThemeTokenProps> = {}
+
+  themeDict.forEach((item) => {
+    const cssVarValue = computedStyle.getPropertyValue(item.cssVarName)
+    if (cssVarValue) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      ;(theme as any)[item.name as string] = cssVarValue
```

Comment:
This will be simpler
```suggestion
     Object.assign(theme, { [item.name]: cssVarValue })
```